dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp

```diff
@@ -170,10 +170,18 @@ GeneratorInterfaceIIDM::getQMax() {
     return generatorIIDM_.minMaxReactiveLimits().max();
   } else if (generatorIIDM_.has_reactiveCapabilityCurve()) {
     double qMax = 0;
-    for (unsigned int i = 0; i <= generatorIIDM_.reactiveCapabilityCurve().size(); ++i) {
-      IIDM::ReactiveCapabilityCurve::point const& current_point = generatorIIDM_.reactiveCapabilityCurve()[i];
-      if (current_point.qmax > qMax)
-        qMax = current_point.qmax;
+    if (- getP() < generatorIIDM_.reactiveCapabilityCurve()[0].p) {
```

Comment:
Please add a const double for -getP() as it is used several times

Reply:
Also, please add an assert that !generatorIIDM_.reactiveCapabilityCurve().empty()

Reply:
Ok, I'll add the const for the -getP().
For the assert, isn't this already done with the generatorIIDM_.has_reactiveCapabilityCurve() ?
Thanks

Reply:
Yes that's correct, but I would add it nevertheless as reading a vector without testing its size can make nasty memory bugs (assert are only tested in debug mode so it will not affect release)

Reply:
Ok perfect